modular_citadel/code/modules/client/preferences.dm

```diff
@@ -3,20 +3,758 @@
+				. += "<b>Breast Shape:</b><a href='?_src_=prefs;preference=breasts_shape;task=input'>[features["breasts_shape"]]</a><br>"
+			/*
+			. += "<h3>Ovipositor</h3>"
+			. += "<b>Has Ovipositor:</b><a href='?_src_=prefs;preference=has_ovi'>[features["has_ovi"] == TRUE ? "Yes" : "No"]</a>"
+			if(features["has_ovi"])
+				. += "<b>Ovi Color:</b><span style='border: 1px solid #161616; background-color: #[features["ovi_color"]];'>&nbsp;&nbsp;&nbsp;</span> <a href='?_src_=prefs;preference=ovi_color;task=input'>Change</a>"
+				. += "<h3>Eggsack</h3>"
+				. += "<b>Has Eggsack:</b><a href='?_src_=prefs;preference=has_eggsack'>[features["has_eggsack"] == TRUE ? "Yes" : "No"]</a><BR>"
+				if(features["has_eggsack"] == TRUE)
+					. += "<b>Color:</b><span style='border: 1px solid #161616; background-color: #[features["eggsack_color"]];'>&nbsp;&nbsp;&nbsp;</span> <a href='?_src_=prefs;preference=eggsack_color;task=input'>Change</a>"
+					. += "<b>Egg Color:</b><span style='border: 1px solid #161616; background-color: #[features["eggsack_egg_color"]];'>&nbsp;&nbsp;&nbsp;</span> <a href='?_src_=prefs;preference=egg_color;task=input'>Change</a>"
+					. += "<b>Egg Size:</b><a href='?_src_=prefs;preference=egg_size;task=input'>[features["eggsack_egg_size"]]\" Diameter</a>"
+			. += "</td>"
+			*/
+			. += "</td></tr></table>"
+
+	if(current_tab == 3)
+		if(!gear_tab)
+			gear_tab = GLOB.loadout_items[1]
+		. += "<table align='center' width='100%'>"
+		. += "<tr><td colspan=4><center><b><font color='[gear_points == 0 ? "#E67300" : "#3366CC"]'>[gear_points]</font> loadout points remaining.</b> \[<a href='?_src_=prefs;preference=gear;clear_loadout=1'>Clear Loadout</a>\]</center></td></tr>"
+		. += "<tr><td colspan=4><center>You can only choose one item per category, unless it's an item that spawns in your backpack or hands.</center></td></tr>"
+		. += "<tr><td colspan=4><center><b>"
+		var/firstcat = TRUE
+		for(var/i in GLOB.loadout_items)
+			if(firstcat)
+				firstcat = FALSE
+			else
+				. += " |"
+			if(i == gear_tab)
+				. += " <span class='linkOn'>[i]</span> "
+			else
+				. += " <a href='?_src_=prefs;preference=gear;select_category=[i]'>[i]</a> "
+		. += "</b></center></td></tr>"
+		. += "<tr><td colspan=4><hr></td></tr>"
+		. += "<tr><td colspan=4><b><center>[gear_tab]</center></b></td></tr>"
+		. += "<tr><td colspan=4><hr></td></tr>"
+		. += "<tr style='vertical-align:top;'><td width=15%><b>Name</b></td>"
+		. += "<td width=5% style='vertical-align:top'><b>Cost</b></td>"
+		. += "<td><font size=2><b>Restrictions</b></font></td>"
+		. += "<td><font size=2><b>Description</b></font></td></tr>"
+		for(var/j in GLOB.loadout_items[gear_tab])
+			var/datum/gear/gear = GLOB.loadout_items[gear_tab][j]
+			var/donoritem
+			if(gear.ckeywhitelist && gear.ckeywhitelist.len)
+				donoritem = TRUE
+				if(user && user.client && user.client.ckey && !(gear.ckeywhitelist.Find(user.client.ckey)))
+					continue
```

Comment:
This continue doesn't actually stop donor items from being shown to people without donor ckeys, it needs to be something else.